sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveWriterContainers.scala

```diff
@@ -234,7 +234,13 @@ private[spark] class SparkHiveDynamicPartitionWriterContainer(
         val outputPath = FileOutputFormat.getOutputPath(conf.value)
         assert(outputPath != null, "Undefined job output-path")
         val workPath = new Path(outputPath, dynamicPartPath.stripPrefix("/"))
-        new Path(workPath, getOutputName)
+        val path = new Path(workPath, getOutputName)
+        val fs = path.getFileSystem(conf.value)
+        // this judgement is added for SPARK-6067
+        if (fs.exists(path) && path.toUri.toString.indexOf("part-") > 0) {
```

Comment:
Maybe check for `part-` before getting a file system or checking if the file exists. This saves RPCs.
Did you mean `> 0` in the sense of `contains`? that's not quite correct if so, since the condition should be `>= 0`. But then `contains` would be simpler anyway.
Finally, duplicat -> duplicate